src/components/Pages/index.js

```diff
@@ -23,52 +21,54 @@ const VmsPage = () => {
 class VmDetailPage extends React.Component {
   constructor (props) {
     super(props)
-    this.requestSent = false
-  }
-
-  componentWillMount () {
-    if (Selectors.isFilterChecked()) {
-      this.props.getVms({ vmId: this.props.match.params.id })
+    this.state = {
+      vmId: undefined,
     }
   }
 
-  componentWillUpdate () {
-    const vmInStore = this.props.vms.getIn(['vms', this.props.match.params.id])
-    if (!vmInStore && Selectors.isFilterChecked() && !this.requestSent) {
-      this.requestSent = true
-      this.props.getVms({ vmId: this.props.match.params.id })
+  static getDerivedStateFromProps (props, state) {
+    if (state.vmId !== props.match.params.id) {
```

Comment:
I think you should add one more condition to avoid sending request repeatedly.

Reply:
I'm not sure what extra condition you have in mind.  As it is, a request will only be fired if the match id (VM UUID or Pool UUID) is changed.  That is only changed if the user navigates to a new a page either by clicking a link or a left hand navigation list item.  

Even if other properties change, and this function is called repeatedly, the prop id check against state id will prevent requesting the same id multiple times.

Reply:
Hmm... ok.